### PR TITLE
chore: unified slog message formatting to improve logging consistency

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at 2f1fa2e</samp>

This pull request improves the logging of various `emqx` applications and modules by refactoring the log messages to use structured logging macros, atoms, and descriptive field names. It also fixes some typos and formatting errors in the logs and comments, and increments the version numbers of the affected applications.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Comment by qzhuyan** [comment]
heavy memory footprint if we use strings, could we align them with atoms instead?

File: apps/emqx/src/emqx_quic_connection.erl [hunk 1]
```diff
@@ -234,7 +234,7 @@ streams_available(_C, {BidirCnt, UnidirCnt}, S) ->
 -spec peer_needs_streams(quicer:connection_handle(), undefined, cb_state()) -> cb_ret().
 peer_needs_streams(_C, undefined, S) ->
     ?SLOG(info, #{
-        msg => "ignore: peer need more streames", info => maps:with([conn_pid, ctrl_pid], S)
+        msg => "ignore_peer_need_more_streames", info => maps:with([conn_pid, ctrl_pid], S)
```
Comment (paulozulato) [inline apps/emqx/src/emqx_quic_connection.erl:237]:
```suggestion
        msg => "ignore_peer_need_more_streams", info => maps:with([conn_pid, ctrl_pid], S)
```

**Review by paulozulato — DISMISSED** [review]

File: apps/emqx/src/emqx_quic_connection.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx/src/emqx_quic_connection.erl:237]: Tiny nit:
```suggestion
        msg => "ignore_peer_needs_more_streams", info => maps:with([conn_pid, ctrl_pid], S)
```
?

**Comment by zhongwencool** [comment]
> heavy memory footprint if we use strings, could we align them with atoms instead?

Is there any information I can learn about string memory? I think if use atom, every time you record, when formatting in this process, it will call atom_to_binary. I am curious why this generated binary consumes less memory than using string directly.

File: apps/emqx_modules/src/emqx_rewrite.erl [hunk 2]
```diff
@@ -91,7 +91,10 @@ register_hook(Rules) ->
         [] ->
             ok;
         _ ->
-            ?SLOG(error, #{rewrite_rule_re_complie_failed => ErrRules}),
+            ?SLOG(error, #{
+                msg => "rewrite_rule_re_complie_failed",
```
Comment (thalesmg) [inline apps/emqx_modules/src/emqx_rewrite.erl:95]:
```suggestion
                msg => "rewrite_rule_re_compile_failed",
```

File: apps/emqx_oracle/src/emqx_oracle.erl [hunk 3]
```diff
@@ -358,7 +358,7 @@ init_prepare(State = #{prepare_sql := Prepares, pool_name := PoolName, params_to
             State#{prepare_sql := Sts};
         Error ->
             LogMeta = #{
-                msg => <<"Oracle Database init prepare statement failed">>, error => Error
+                msg => <<"0racle_init_prepare_statement_failed">>, error => Error
```
Comment (thalesmg) [inline apps/emqx_oracle/src/emqx_oracle.erl:361]:
```suggestion
                msg => <<"oracle_init_prepare_statement_failed">>, error => Error
```

**Comment by zhongwencool** [comment]
@thalesmg 👏👏👏👏👏

**Comment by keynslug** [comment]
Decided to chime in.

> Is there any information I can learn about string memory?

String is a list of chars, which means each element costs at least 1 word for a char + at 1 for a cons cell. See also this [chapter](https://www.erlang.org/doc/efficiency_guide/advanced.html#memory).

> heavy memory footprint if we use strings

However, I doubt this is applicable to the current PR. All the strings are constant literals, so they are part of literal heap already, they should not cost any allocations per `?SLOG(...)` invocation. Considering that Erlang IO/term formatting works with strings internally anyway, it's likely that the opposite will happen: formatting an atom boils down to `atom_to_list/1`, and thus will cause an allocation on the process heap, while formatting a string will not.

I'm not 100% sure about that though, take with a grain of salt. Just wanted to highlight that this rule ("strings = more wasted memory") might not work in this specific case.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**